src/awx_plugins/credentials/azure_kv.py

```diff
@@ -71,24 +88,52 @@
     ],
     'required': [
         'url',
-        'client',
-        'secret',
-        'tenant',
         'secret_field',
     ],
 }
 
 
-def azure_keyvault_backend(**kwargs):
-    csc = ClientSecretCredential(
-        tenant_id=kwargs['tenant'],
-        client_id=kwargs['client'],
-        client_secret=kwargs['secret'],
-    )
-    kv = SecretClient(credential=csc, vault_url=kwargs['url'])
-    return kv.get_secret(
-        name=kwargs['secret_field'],
-        version=kwargs.get('secret_version', ''),
+def azure_keyvault_backend(
+        url: str,
```

Comment:
Let's make all the args keyword-only since that's the only way awx invokes this. Plus, it's usually bad to allow positional args when there's many.
```suggestion
        *,
        url: str,
```